### PR TITLE
fix-uncaught-error-process-exits

### PR DESCRIPTION
**Pull request by evoxtorm**

### CODE REVIEW

File: lib/utils/shell.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const { spawn } = require('child_process')
 const { tab } = require('../utils')
+const { setWorfklowStatusPermanentFailure } = require('../reducers/workflowState.js')
```
Comment (thejmazz) [inline lib/utils/shell.js:5]: remove

File: lib/utils/shell.js [hunk 2]
```diff
@@ -31,16 +32,16 @@ const shell = (cmd, opts = {}, logger) => {
   }
 
   function onClose(code) {
-    // console.log('CP closed: ' + code)
+     //console.log('CP closed: ' + code)
     if (code !== 0) {
       logger.emit('log', `Child process exiting with code ${code}`)
       logger.emit('log', 'Exiting...')
-      setTimeout(() => process.exit(code), 5000)
+       //store.dispatch({ type: 'SET_WORKFLOW_STATUS', payload: 'permanentFailure' })
```
Comment (thejmazz) [inline lib/utils/shell.js:39]: remove, take out the `setTimeout(() => process.exit(code), 5000)` as well

File: examples/pipelines/capitalize/wait-error.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+
+console.log('Starting')
```
Comment (thejmazz) [inline examples/pipelines/capitalize/wait-error.js:3]: formatting

File: lib/utils/shell.js (hunk 2 above)
Comment (thejmazz) [reply lib/utils/shell.js:39]: also, it is obvious this will fail as `store` is not imported

**Review by thejmazz — CHANGES_REQUESTED** [review]
dont commit the graphson.json files, and close this PR and make a new one merging into develop, with the changes from comments.

also maybe make a new folder in `examples/pipelines` rather than putting things in capitalized

**Comment by evoxtorm** [comment]
Ok I'll make changes

**Pull request closed without merging**